src/Compilers/CSharp/Portable/Symbols/ReducedExtensionMethodSymbol.cs

```diff
@@ -34,13 +34,15 @@ internal sealed class ReducedExtensionMethodSymbol : MethodSymbol
         /// </summary>
         /// <param name="compilation">Compilation used to check constraints.
         /// The latest language version is assumed if this is null.</param>
-        public static MethodSymbol Create(MethodSymbol method, TypeSymbol receiverType, CSharpCompilation compilation)
+        public static MethodSymbol Create(MethodSymbol method, TypeSymbol receiverType, CSharpCompilation compilation, bool discardUseSiteInfo = false)
         {
             Debug.Assert(method.IsExtensionMethod && method.MethodKind != MethodKind.ReducedExtension);
             Debug.Assert(method.ParameterCount > 0);
             Debug.Assert((object)receiverType != null);
 
-            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.DiscardedDependencies;
+            var useSiteInfo = discardUseSiteInfo
+                ? CompoundUseSiteInfo<AssemblySymbol>.Discarded
+                : CompoundUseSiteInfo<AssemblySymbol>.DiscardedDependencies;
```

Comment:
This may have a behavioral change, but let's see if it will be captured by tests and check whether it's an important scenario for LookupSymbols or not.